src/Platforms/AbstractPlatform.php

```diff
@@ -2365,25 +2377,33 @@ public function getCheckDeclarationSQL(array $definition)
      * Obtains DBMS specific SQL code portion needed to set a unique
      * constraint declaration to be used in statements like CREATE TABLE.
      *
-     * @param string $name  The name of the unique constraint.
-     * @param Index  $index The index definition.
+     * @param string           $name       The name of the unique constraint.
+     * @param UniqueConstraint $constraint The unique constraint definition.
      *
      * @return string DBMS specific SQL code portion needed to set a constraint.
      *
      * @throws InvalidArgumentException
      */
-    public function getUniqueConstraintDeclarationSQL($name, Index $index)
+    public function getUniqueConstraintDeclarationSQL($name, UniqueConstraint $constraint)
     {
-        $columns = $index->getColumns();
+        $columns = $constraint->getQuotedColumns($this);
         $name    = new Identifier($name);
 
         if (count($columns) === 0) {
             throw new InvalidArgumentException("Incomplete definition. 'columns' required.");
         }
 
-        return 'CONSTRAINT ' . $name->getQuotedName($this) . ' UNIQUE ('
-            . $this->getIndexFieldDeclarationListSQL($index)
-            . ')' . $this->getPartialIndexSQL($index);
+        $flags = ['UNIQUE'];
+
+        if ($constraint->hasFlag('clustered')) {
+            $flags[] = 'CLUSTERED';
+        }
+
+        $constraintName  = $name->getQuotedName($this);
+        $constraintName  = ! empty($constraintName) ? $constraintName . ' ' : '';
```

Comment:
The trailing whitespace is not part of the constraint name. Would it make sense to rename `$flags` to something like `$tokens`, push the name there and then join everything?

Reply:
Please avoid `empty()`. It's forbidden by PHPStan in master.